samplesheets/tests/test_permissions_api_taskflow.py

```diff
@@ -74,3 +83,329 @@ def test_get_archive(self):
+        self.investigation = self.import_isa_from_file(SHEET_PATH, self.project)
+        self.study = self.investigation.studies.first()
+        self.assay = self.study.assays.first()
+
+        # Set up iRODS data
+        self.make_irods_colls(self.investigation)
+        self.assay_path = self.irods_backend.get_path(self.assay)
+        self.path = os.path.join(self.assay_path, TEST_FILE_NAME)
+        self.path_md5 = os.path.join(self.assay_path, f'{TEST_FILE_NAME}.md5')
+        # Create objects
+        self.file_obj = self.irods.data_objects.create(self.path)
+        self.md5_obj = self.irods.data_objects.create(self.path_md5)
+
+
+class TestIrodsRequestCreateAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for IrodsRequestCreateAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        # Set up URLs
+        self.url = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        # Set up post data
+        self.post_data = {'path': self.path + '/', 'description': 'bla'}
+
+    def test_create(self):
+        """Test post() in IrodsRequestCreateAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,  # Inherited owner
+            self.user_owner,
+            self.user_delegate,
+            self.user_contributor,
+        ]
+        bad_users = [self.user_guest, self.user_no_roles]
+
+        self.assert_response_api(
+            self.url, good_users, 200, method='POST', data=self.post_data
+        )
+        self.assert_response_api(
+            self.url, bad_users, 403, method='POST', data=self.post_data
+        )
+
+    @override_settings(PROJECTROLES_ALLOW_ANONYMOUS=True)
+    def test_create_anon(self):
+        """Test post() in IrodsRequestCreateAPIView with anonymous access"""
+        self.project.set_public()
+        self.assert_response_api(
+            self.url,
+            self.anonymous,
+            401,
+            method='POST',
+            data=self.post_data,
+        )
+
+
+class TestIrodsRequestUpdateAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for IrodsRequestUpdateAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        # Set up URLs
+        self.url_create = reverse(
+            'samplesheets:api_irods_request_create',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        # Set up post data
+        self.post_data = {'path': self.path + '/', 'description': 'bla'}
+        self.update_data = {'path': self.path, 'description': 'Updated'}
+
+    def test_update(self):
+        """Test post() in IrodsRequestUpdateAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,  # Inherited owner
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [self.user_guest, self.user_no_roles, self.user_contributor]
```

Comment:
Let's maintain a consistent ordering of users so it's easier to keep track of these tests. Contributor should come before guest.